content/4_Gold/Combinatorics.mdx

```diff
@@ -903,49 +920,174 @@ For example, let $X$ be the probability distribution of a fair 6-sided die. $P(X
 
 ### Explanation
 
-Let $X$ represent the probability distribution of maximum number of candies a child gets. To get $E[X]$, we just need $P(X = x)$ for $1 \leq x \leq k$. $(\frac{x}{k})^n$ is the probability that each child gets at most $x$ candies. To get $P(X = x)$, we must subtract out the probability that each child gets strictly less than $x$ candies, which is $(\frac{x - 1}{k})^n$.
+Let $X$ represent the probability distribution of maximum number of candies a child gets. To get $E[X]$, we  need $P(X = x)$ for $1 \leq x \leq k$. $(\frac{x}{k})^n$ is the probability that each child gets at most $x$ candies. To get $P(X = x)$, we must subtract out the probability that each child gets strictly less than $x$ candies, which is $(\frac{x - 1}{k})^n$.
 
-Therefore, $P(X = x) = (\frac{x}{k})^n - (\frac{x - 1}{k})^n$, allowing us to easily calculate $E[X]$.
+Therefore, $P(X = x) = (\frac{x}{k})^n - (\frac{x - 1}{k})^n$, allowing us to calculate $E[X]$.
 
 ### Implementation
 
-**Time Complexity:** $\mathcal{O}(nk)$ using naive exponentiation or $\mathcal{O}(k \log n)$ using binary exponentiation or std::pow.
+**Time Complexity:** $\mathcal{O}(nk)$ using naive exponentiation or $\mathcal{O}(k \log n)$ using binary exponentiation.
+
+<LanguageSection>
+<CPPSection>
 
 ```cpp
-#include <bits/stdc++.h>
-using namespace std;
-int n, k;
-void solve() {
-	cin >> n >> k;
-	double out = 0;
-	for (int i = 1; i <= k; i++) {
-		out += i * (pow((double)i / k, n) - pow((double)(i - 1) / k, n));
-	}
+#include <cmath>
+#include <iostream>
+#include <iomanip>
 
-	// set output precision to 6 decimal points
-	cout << setprecision(6) << fixed;
-	cout << out << "\n";
-}
+using std::cout;
+using std::endl;
+
+int main() {
+    int n;
+    int k;
+    std::cin >> n >> k;
+
+    double expect_max = 0;
+    for (int i = 1; i <= k; i++) {
+        expect_max += i * (pow((double)i / k, n) - pow((double)(i - 1) / k, n));
+    }
 
-int32_t main() {
-	std::ios::sync_with_stdio(false);
-	cin.tie(NULL);
-	solve();
+    cout << std::setprecision(6) << std::fixed;  // set output precision
+    cout << expect_max << endl;
 }
 ```
 
 </CPPSection>
 <PySection>
 
 ```py
-from math import comb
+n, k = [int(i) for i in input().split()]
 
-ans = []
-for _ in range(int(input())):
-	marble_num, color_num = [int(i) for i in input().split()]
-	ans.append(str(comb(marble_num - 1, color_num - 1)))
+expected_max = 0
+for c in range(1, k + 1):
+    expected_max += c * ((c / k) ** n - ((c - 1) / k) ** n)
 
-print("\n".join(ans))
+print(f"{expected_max:.6f}")
+```
+
+</PySection>
+</LanguageSection>
+
+## Linearity of Expectation
```

Comment:
would be cool to link some resources for this section since the explanation is pretty brief.

https://brilliant.org/wiki/linearity-of-expectation/ - off the top of my head, brilliant has an article which is pretty good